src/robusta/integrations/mail/sender.py

```diff
@@ -0,0 +1,198 @@
+    return obj
+
+
+class MailTransformer(Transformer):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.file_blocks = []
+
+    def block_to_html(self, block: BaseBlock) -> str:
+        # TODO should we additionally support ScanReportBlock here?
+        if isinstance(block, FileBlock):
+            self.file_blocks.append(block)
+            return f"<p>See attachment {block.filename}</p>"
+        elif isinstance(block, LinksBlock):
+            if getattr(block, "html_class", None):
+                class_part = f' class="{block.html_class}"'
+            else:
+                class_part = ""
+            return (
+                f"<ul{class_part}>\n"
+                + "\n".join(f'  <li><a href="{link.url}">{link.text}</a></li>' for link in block.links)
+                + "\n</ul>\n"
+            )
+        else:
+            return super().block_to_html(block)
+
+
+class MailSender:
+    def __init__(self, mailto: str, account_id: str, cluster_name: str, signing_key: str):
+        self.mailto = mailto
+        self.signing_key = signing_key
+        self.account_id = account_id
+        self.cluster_name = cluster_name
+
+    def send_finding_via_email(self, finding: Finding, platform_enabled: bool):
+        blocks: List[BaseBlock] = []
+
+        if finding.title:
+            status: FindingStatus = (
+                FindingStatus.RESOLVED if finding.title.startswith("[RESOLVED]") else FindingStatus.FIRING
+            )
+            blocks.append(self.__create_finding_header(finding, status))
+        else:
+            # TODO is this correct? Is it possible for the finding to have no title at all?
+            status: FindingStatus = FindingStatus.FIRING
+
+        if platform_enabled:
+            blocks.append(self.__create_links(finding, html_class="header_links"))
+
+        blocks.append(MarkdownBlock(text=f"*Source:* `{self.cluster_name}`"))
+        if finding.description:
+            blocks.append(MarkdownBlock(f"{Emojis.Alert.value} *Alert:* {finding.description}"))
+
+        for enrichment in finding.enrichments:
+            blocks.extend(enrichment.blocks)
+
+        transformer = MailTransformer()
+        html_body = self.__build_html(transformer.to_html(blocks).strip())
+
+        ap_obj = apprise.Apprise()
+        attachments = apprise.AppriseAttachment()
+        attachment_files = []
+        try:
+            for file_block in transformer.file_blocks:
+                # This is awkward, but it's the standard way to handle
+                # attachments in apprise - by providing local filesystem
+                # names.
+                f = tempfile.NamedTemporaryFile()
+                attachment_files.append(f)
+                f.write(file_block.contents)
+                attachments.add(f.name)
+            ap_obj.add(self.mailto)
+            logging.info(f"MailSender: sending title={finding.title}, body={html_body}")
```

Comment:
This might be too spammy, maybe change it to debug?

Reply:
changed